pkg/webhook/injection.go

```diff
@@ -63,23 +82,57 @@ func (si *SidecarInjector) supportsNativeSidecar() (bool, error) {
 	return supportsNativeSidecar, nil
 }
 
-func injectSidecarContainer(pod *corev1.Pod, config *Config, supportsNativeSidecar bool) {
-	nativeSidecarEnabled := true
-	if enable, ok := pod.Annotations[GcsFuseNativeSidecarEnableAnnotation]; ok {
-		parsedAnnotation, err := ParseBool(enable)
-		if err != nil {
-			klog.Errorf("failed to parse enableNativeSidecar annotation... ignoring annotation: %v", err)
-		} else {
-			nativeSidecarEnabled = parsedAnnotation
-		}
-	}
-	if supportsNativeSidecar && nativeSidecarEnabled {
+func injectSidecarContainer(pod *corev1.Pod, config *Config, injectAsNativeSidecar bool) {
+	if injectAsNativeSidecar {
 		pod.Spec.InitContainers = insert(pod.Spec.InitContainers, GetNativeSidecarContainerSpec(config), getInjectIndex(pod.Spec.InitContainers))
 	} else {
 		pod.Spec.Containers = insert(pod.Spec.Containers, GetSidecarContainerSpec(config), getInjectIndex(pod.Spec.Containers))
 	}
 }
 
+func (si *SidecarInjector) injectMetadataPrefetchSidecarContainer(pod *corev1.Pod, config *Config, injectAsNativeSidecar bool) {
```

Comment:
i dont think we should respect the injectAsNativeSidecar parameter. Rather, we should make this decision based on the location of the `gke-gcsfuse-sidecar`

Reply:
since we are injecting in the same flow, injectAsNativeSidecar inherently tells us the location of our fuse sidecar.